Linux Smoke Tests CI failing

Fails during the `Building cp312-manylinux_x86_64 wheel` stage. 

Run on `main`: https://github.com/OasisLMF/OasisLMF/actions/runs/28932990599

Error log: 
```
Processing /project
    Installing build dependencies: started
    Installing build dependencies: finished with status 'error'
    error: subprocess-exited-with-error
    
    × pip subprocess to install build dependencies did not run successfully.
    │ exit code: 1
    ╰─> [51 lines of output]
        Collecting setuptools<77.0,>=61.0
          Using cached setuptools-76.1.0-py3-none-any.whl.metadata (6.7 kB)
        Collecting wheel
          Using cached wheel-0.47.0-py3-none-any.whl.metadata (2.3 kB)
        Collecting Cython>=3.0
          Downloading cython-3.2.8-cp312-cp312-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl.metadata (4.2 kB)
        Collecting numpy>=2.0
          Downloading numpy-2.5.1.tar.gz (20.8 MB)
             ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 20.8/20.8 MB 184.1 MB/s eta 0:00:00
          Installing build dependencies: started
          Installing build dependencies: finished with status 'done'
          Getting requirements to build wheel: started
          Getting requirements to build wheel: finished with status 'done'
          Installing backend dependencies: started
          Installing backend dependencies: finished with status 'done'
          Preparing metadata (pyproject.toml): started
          Preparing metadata (pyproject.toml): finished with status 'error'
          error: subprocess-exited-with-error
        
          × Preparing metadata (pyproject.toml) did not run successfully.
          │ exit code: 1
          ╰─> [19 lines of output]
              + /opt/python/cp312-cp312/bin/python /tmp/pip-install-iwabzb96/numpy_cc5aef4eef244b51b1e4a26c0a7e5c8e/vendored-meson/meson/meson.py setup /tmp/pip-install-iwabzb96/numpy_cc5aef4eef244b51b1e4a26c0a7e5c8e /tmp/pip-install-iwabzb96/numpy_cc5aef4eef244b51b1e4a26c0a7e5c8e/.mesonpy-y8f0mczb -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/tmp/pip-install-iwabzb96/numpy_cc5aef4eef244b51b1e4a26c0a7e5c8e/.mesonpy-y8f0mczb/meson-python-native-file.ini
              The Meson build system
              Version: 1.11.1
              Source dir: /tmp/pip-install-iwabzb96/numpy_cc5aef4eef244b51b1e4a26c0a7e5c8e
              Build dir: /tmp/pip-install-iwabzb96/numpy_cc5aef4eef244b51b1e4a26c0a7e5c8e/.mesonpy-y8f0mczb
              Build type: native build
              Project name: NumPy
              Project version: 2.5.1
              C compiler for the host machine: cc (gcc 10.2.1 "cc (GCC) 10.2.1 20210130 (Red Hat 10.2.1-11)")
              C linker for the host machine: cc ld.bfd 2.35-5
              C++ compiler for the host machine: c++ (gcc 10.2.1 "c++ (GCC) 10.2.1 20210130 (Red Hat 10.2.1-11)")
              C++ linker for the host machine: c++ ld.bfd 2.35-5
              Cython compiler for the host machine: cython (cython 3.2.8)
              Host machine cpu family: x86_64
              Host machine cpu: x86_64
        
              ../meson.build:25:4: ERROR: Problem encountered: NumPy requires GCC >= 10.3
        
              A full log can be found at /tmp/pip-install-iwabzb96/numpy_cc5aef4eef244b51b1e4a26c0a7e5c8e/.mesonpy-y8f0mczb/meson-logs/meson-log.txt
              [end of output]
        
          note: This error originates from a subprocess, and is likely not a problem with pip.
        error: metadata-generation-failed
        
        × Encountered error while generating package metadata.
        ╰─> See above for output.
        
        note: This is an issue with the package mentioned above, not pip.
        hint: See above for details.
        [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
  error: subprocess-exited-with-error
  
  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
                                                             ✕ 13.91s
Error: Command ['python', '-m', 'pip', 'wheel', PurePosixPath('/project'), '--wheel-dir=/tmp/cibuildwheel/built_wheel', '--no-deps'] failed with code 1. 

Error: Process completed with exit code 1.
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -93,7 +93,7 @@ jobs:
     - uses: actions/checkout@v4
 
     - name: Build wheels
-      uses: pypa/cibuildwheel@v2.21.3
+      uses: pypa/cibuildwheel@v4.1.0
       env:
         CIBW_ARCHS: ${{ matrix.archs }}
         CIBW_SKIP: "*musllinux*"

